qirlib/src/module.rs

```diff
@@ -36,21 +36,25 @@ pub(crate) fn load_module_from_bitcode_template<'ctx>(
     Module::parse_bitcode_from_buffer(&buffer, context).map_err(|e| e.to_string())
 }
 
-pub(crate) fn load_module_from_bitcode_file<'ctx, P: AsRef<Path>>(
-    path: P,
-    context: &'ctx inkwell::context::Context,
-) -> Result<Module<'ctx>, String> {
+pub(crate) fn load_module_from_bitcode_file(
+    path: impl AsRef<Path>,
+    context: &'_ inkwell::context::Context,
+) -> Result<Module<'_>, String> {
```

Comment:
I think you can always omit `'_` when there's only a single lifetime parameter.

```suggestion
    context: &inkwell::context::Context,
) -> Result<Module, String> {
```